ui/shared/DropdownMenuItems.jsx

```diff
@@ -4,39 +4,60 @@ import { DropdownMenu, DropdownItem } from 'reactstrap';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
 import { faCheck } from '@fortawesome/free-solid-svg-icons';
 
+const createDropdownItem = (item, namespace, updateData, selectedItem) => {
+  return (
+    <DropdownItem
+      tag="a"
+      key={`${namespace}${item}`}
+      onClick={() => updateData(item)}
+    >
+      <FontAwesomeIcon
+        icon={faCheck}
+        className={`mr-1 ${selectedItem === item ? '' : 'hide'}`}
+        title={selectedItem === item ? 'Checked' : ''}
+      />
+      {item}
+    </DropdownItem>
+  );
+};
+
 const DropdownMenuItems = ({
   selectedItem,
   updateData,
   options,
+  pinned,
   namespace = '',
 }) => (
   <DropdownMenu className="overflow-auto dropdown-menu-height">
-    {options.map(item => (
-      <DropdownItem
-        tag="a"
-        key={`${namespace}${item}`}
-        onClick={() => updateData(item)}
-      >
-        <FontAwesomeIcon
-          icon={faCheck}
-          className={`mr-1 ${selectedItem === item ? '' : 'hide'}`}
-          title={selectedItem === item ? 'Checked' : ''}
-        />
-        {item}
-      </DropdownItem>
-    ))}
+    {/* Items pinned to top of dropdown */}
+    {pinned.length > 0 &&
+      options
```

Comment:
This filters out the pinned items that are not in options, but we also lose the order we defined in pinned.projects (it just orders it alphabetically).

To keep the order of the pinned items as you defined it, it should be the pinned array to be looped:
```
pinned
        .filter(item => options.includes(item))
        .map(item =>
          createDropdownItem(item, namespace, updateData, selectedItem),
        )}
```

Reply:
Should the pinned items be in any particular order? alphabetizing was intentional

Reply:
I'm not sure if we need a particular order :). I think it would be nice to have control over the order in which the pinned values appear and also, when I saw the list of pinned repos defined I assumed that was the final order :D but I'm ok with both options

Reply:
From my discussion with Dave, we need explicit control over the pinned items. So alphabetizing them isn't something we want. We want alphabetizing only for the rest of the items.